pkg/build/builder/util.go

```diff
@@ -26,3 +30,67 @@ func getBuildEnvVars(build *api.Build) map[string]string {
 	}
 	return envVars
 }
+
+type BuildConfigClientInterface interface {
+	GetBuildConfig(namespace, name string) (*buildapi.BuildConfig, error)
+	UpdateBuildConfig(buildConfig *buildapi.BuildConfig) error
+}
+
+type BuildClientInterface interface {
+	CreateBuild(namespace string, build *buildapi.Build) (*buildapi.Build, error)
+	CreateBuildFromConfig(config *buildapi.BuildConfig, revision *buildapi.SourceRevision, imageSubstitutions map[string]string) error
+	CreateBuildWithImmutableId(config *buildapi.BuildConfig) error
+	UpdateBuild(namespace string, build *buildapi.Build) (*buildapi.Build, error)
+}
+
+// OSClientBuildClient is a BuildClientInterface which delegates to the OpenShift client interfaces
+type OSClientBuildClient struct {
```

Comment:
(to be clear i'm referring to the OSClientBuildClient type, now i'm not sure which type you were referring to.  if it was the BuildClientInterface, i guess we'll see if my round 2 updates after discussing w/ @mfojtik and @soltysh go over any better.